homeassistant/components/p1_monitor/const.py

```diff
@@ -0,0 +1,221 @@
+"""Constants for the P1 Monitor integration."""
+from __future__ import annotations
+
+from datetime import timedelta
+import logging
+from typing import Final
+
+from homeassistant.components.sensor import (
+    STATE_CLASS_MEASUREMENT,
+    STATE_CLASS_TOTAL_INCREASING,
+    SensorEntityDescription,
+)
+from homeassistant.const import (
+    CURRENCY_EURO,
+    DEVICE_CLASS_CURRENT,
+    DEVICE_CLASS_ENERGY,
+    DEVICE_CLASS_GAS,
+    DEVICE_CLASS_POWER,
+    DEVICE_CLASS_VOLTAGE,
+    ELECTRIC_CURRENT_AMPERE,
+    ELECTRIC_POTENTIAL_VOLT,
+    ENERGY_KILO_WATT_HOUR,
+    POWER_WATT,
+    VOLUME_CUBIC_METERS,
+)
+
+DOMAIN: Final = "p1_monitor"
+LOGGER = logging.getLogger(__package__)
+SCAN_INTERVAL = timedelta(seconds=5)
+
+ATTR_ENTRY_TYPE: Final = "entry_type"
+ENTRY_TYPE_SERVICE: Final = "service"
+
+SERVICE_SMARTMETER: Final = "smartmeter"
+SERVICE_PHASES: Final = "phases"
+SERVICE_SETTINGS: Final = "settings"
+
+SERVICES: dict[str, str] = {
+    SERVICE_SMARTMETER: "SmartMeter",
+    SERVICE_PHASES: "Phases",
+    SERVICE_SETTINGS: "Settings",
+}
+
+SENSORS: dict[str, tuple[SensorEntityDescription, ...]] = {
```

Comment:
I'd move this to the sensor platform since it describes details of the sensor platform.

Reply:
Personally, I actually like it better if it stays in const 🤔

Reply:
Why? It makes it harder to follow the sensor platform code and also leaks details from the sensor platform that doesn't belong anywhere else.